sample.c

```diff
@@ -39,35 +39,58 @@ long read_file(char const *path, char **buf) {
   return offset;
 }
 
-int melt(char* buf_in, size_t buf_in_len) {
+int melt(char *buf_in, size_t buf_in_len) {
   char *out_buf;
   size_t out_size;
-  char res = rakaly_eu4_melt(buf_in, buf_in_len, &out_buf, &out_size);
-  if (res != 0) {
-    perror("unable to melt save");
+  MeltedBuffer *melt = rakaly_eu4_melt(buf_in, buf_in_len);
+  if (rakaly_melt_error_code(melt)) {
```

Comment:
Perhaps also call rakaly_free_melt in this if block? Sure, the memory leak is very small in this case, as there's no data, but it's still there.